game_tic_tac_toe.py

```diff
@@ -159,11 +184,130 @@ def player_won(self, check_disable):
+            # defines current and next player
+            current_player = self.player_x
+            next_player = self.player_o
+        else: 
+            self.game_btns[move].config(state="disabled", text="O")
+            # adds move to set of moves
+            self.disable_numbers_o.add(move)
+            check_disable = self.disable_numbers_o
+            # defines current and next player
+            current_player = self.player_o
+            next_player = self.player_x
+        
+        self.after_move(check_disable, current_player, next_player)
+
+    def deciding_computer_move(self, comp_symbol):
+        '''In this function the computer decides the next move
+        Parameters:
+        disable_numbers_o_copy(set): copy of the moves up to this moment of the computer
+        disable_numbers_x_copy(set): copy of the moves up to this moment of the player
+        Return
+        move(int): position on the board'''
+        disable_numbers_o_copy = self.disable_numbers_o.copy()
+        disable_numbers_x_copy = self.disable_numbers_x.copy()
+
+        available_moves = []
+        for i in range(9):
+            if self.game_btns[i]['state'] == 'normal':
+                available_moves.append(i)
+        print(f'available moves for the computer {available_moves}')
+        
+        # identify computer player and move done by him 
+        # and by its opponent
+        if comp_symbol == "x":
+            computer_set = disable_numbers_x_copy
+            oponent_set = disable_numbers_o_copy
+        else:
+            computer_set = disable_numbers_o_copy
+            oponent_set = disable_numbers_x_copy
+        print(f'computer already played - disable numbers {computer_set}')
+
+        # Check for every available move if computer can win with the next move
+        # if it can win self.move has a value
+        for i in available_moves:
+            computer_set.add(i)
+            if self.player_won(computer_set):
+                move = i
+                print(f'computer can win!! {move}')
+                return move
+            computer_set.remove(i)
+        print('computer does not have a chance to win')
+
+        # Check for every available move if the human can win with the next move
+        # if the human could win next round - the computer will block him/her
+        for i in available_moves:
+            oponent_set.add(i)
+            if self.player_won(oponent_set):
+                move = i
+                print(f'opponent can win!! block: {move}')
+                return move
+            oponent_set.remove(i)
+        print('oponent has no chance to win')
+
+        # Check if any of the corners and center are available to make a move
+        # Center and corners are python-positions [0,2,6,8,4] in the board
+        corners_center_available = []
+        for i in available_moves:
+            if i in [0, 2, 4, 6, 8]:
+                corners_center_available.append(i)
```

Comment:
Try set intersection: `corners_center_available = {0, 2, 4, 6, 8}.intersection(available_moves)`
The parameter before the `.intersection` has to be a set but the argument can also be a list.